app/javascript/components/redfish-server-firmware-update-dialog.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import FirmwareUpdate from "./forms/firmware-update"
+import { handleApiError, fetchFirmwareBinariesForServer, createFirmwareUpdateRequest } from "../utils/api";
+
+const RedfishServerFirmwareUpdateDialog = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [physicalServerIds, setPhysicalServerIds] = useState([]);
+  const [firmwareBinaries, setFirmwareBinaries] = useState([]);
+  const [error, setError] = useState('');
+  const [values, setValues] = useState({});
+
+  const selectedPhysicalServers = () => {
+    if(ManageIQ.gridChecks.length > 0){ // Multi-record page
+      setPhysicalServerIds(ManageIQ.gridChecks);
+    } else if(ManageIQ.record.recordId){ // Single-record page
+      setPhysicalServerIds([ManageIQ.record.recordId]);
+    } else{
+      setError(__('Please select at least one physical server to update firmware for.'));
+      setPhysicalServerIds([]);
+    }
+  };
+
+  const fwBinaryToSelectOption = fwBinary => { return { value: fwBinary.id, label: `${fwBinary.name} (${fwBinary.description})` } };
+
+  const fetchServers = () => fetchFirmwareBinariesForServer(physicalServerIds[0]).then((fwBinaries) => {
+    setFirmwareBinaries(fwBinaries.resources.map(fwBinaryToSelectOption));
+    setLoading(false);
+  }, handleApiError(this));
+
+  const handleFormStateUpdate = (formState) => {
+    props.dispatch({
+      type: "FormButtons.saveable",
+      payload: formState.valid
+    });
+    props.dispatch({
+      type: "FormButtons.pristine",
+      payload: formState.pristine
+    });
+    setValues(formState.values);
+  };
+
+  useEffect(() => {
+    props.dispatch({
+      type: "FormButtons.init",
+      payload: {
+        newRecord: true,
+        pristine: true,
+        addClicked: () => createFirmwareUpdateRequest(physicalServerIds, values.firmwareBinary)
```

Comment:
Hey @Hyperkid123 any idea why this lambda (executed when I press Submit button on the popup) always gets initial (i.e. empty) value for both states, `physicalServerIds = []` and `values = {}`?